src/resolv/lookup/addr.rs

```diff
@@ -137,3 +138,31 @@ fn hexdigit(nibble: u8) -> u8 {
         _ => unreachable!(),
     }
 }
+
+//============ Tests =========================================================
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use core::str::FromStr;
+
+    #[test]
+    fn test_dname_from_addr() {
+        assert_eq!(
+            dname_from_addr([192, 0, 2, 12].into()),
+            Dname::<Octets128>::from_str("12.2.0.192.in-addr.arpa").unwrap()
+        );
+        assert_eq!(
+            dname_from_addr(
+                [0x2001, 0xdb8, 0x2c88, 0x0, 0xfff, 0x1, 0x1200, 0xc032]
```

Comment:
could maybe the tested address contain all hex digits possible? Including fixed 9?

Reply:
It should, yes!